flux-plugin/src/flux-applications/application.tsx

```diff
@@ -0,0 +1,390 @@
+import {
+  ConditionsTable,
+  DateLabel,
+  Link,
+  MainInfoSection,
+  SectionBox,
+  Table,
+} from '@kinvolk/headlamp-plugin/lib/components/common';
+import { useLocation } from 'react-router';
+import Event from '@kinvolk/headlamp-plugin/lib/K8s/event';
+import React from 'react';
+import { K8s } from '@kinvolk/headlamp-plugin/lib';
+import { GraphView } from '../resoureceMap/GraphView';
+import { GraphSource } from '../resoureceMap/GraphModel';
+import { WorkloadsSource } from '../resoureceMap/sources/WorkloadSource';
+import {
+  SuspendAction,
+  ResumeAction,
+  SyncWithSourceAction,
+  SyncWithoutSourceAction,
+} from '../actions/index';
+import { KubeObject } from '@kinvolk/headlamp-plugin/lib/lib/k8s/cluster';
+import YAML from 'yaml';
+import { getSourceNameAndType } from '../helpers/index';
+import Editor from '@monaco-editor/react';
+
+const KUSTOMIZATION_API_VERSION = 'kustomize.toolkit.fluxcd.io';
+const HELM_API_VERSION = 'helm.toolkit.fluxcd.io';
+
+const KUSTOMIZE_CRD = 'kustomizations.kustomize.toolkit.fluxcd.io';
+const HELMRELEASE_CRD = 'helmreleases.helm.toolkit.fluxcd.io';
+
+function GetResourcesFromInventory(props: {
+  inventory: {
+    id: string;
+    v: string;
+  }[];
+}) {
+  const [resources, setResources] = React.useState([]);
+  function parseID(id: string) {
+    /* ID is the string representation of the Kubernetes resource object's
+    metadata,
+    in the format '<namespace>_<name>_<group>_<kind>'.
+    */
+    const parsedID = id.split('_');
+    const namespace = parsedID[0];
```

Comment:
Not all resources have a namespace, I suggest testing this on cluster where you've run `flux bootstrap` and see if the `flux-system` Kustomization renders here the ClusterRoles, Namespaces, etc.

Reply:
The K8s API we use handles this, but i was missing passing an undefined namespace which was an issue, thanks @stefanprodan for pointing this out.